uefi/firmwarevolume.go

```diff
@@ -189,6 +190,39 @@ func (fv *FirmwareVolume) Assemble() ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
+	// Make sure we don't go over the size.
+	fvLen := uint64(len(fv.buf))
+	fOffset := fv.DataOffset
+	for _, f := range fv.Files {
+		fBuf, err := f.Assemble()
+		if err != nil {
+			return nil, err
+		}
+		fLen := uint64(len(fBuf))
+		if fLen+fOffset > fvLen {
+			// TODO: Actually loop through and calculate the full size so we know how much to reduce by.
+			// For now we just return early
+			return nil, errors.New("insufficient space in FV, files too big")
```

Comment:
Can you include the FV size in the diagnosis message? Something like:

    insufficient space in 43.5KiB FV, files too big

Reply:
done